overview/glossary.markdown

```diff
@@ -41,74 +102,177 @@ Used to restrict when / where promises are evaluated.
 Class guards are sometimes called context class expressions.
 
-#### CMDB
+#### Client
+
+In traditional computer networks and software, the client is the program which connects to a server, i.e. the software which initiates the connection in a networked system.
+We say that a server is listening for incoming connections, and servers frequently serve thousands or even millions of clients simultaneously.
+
+In CFEngine, we use the word client to describe all of the hosts which are not hubs.
+A CFEngine hub runs a policy server, which all clients connect to in order to fetch policy.
+
+Historically, the term agent has sometimes been used for this same meaning.
+However, agent also refers to the agent component (the `cf-agent` binary), and thus, when discussing the role of a CFEngine host, _client_ is the preferred term for these hosts which are not hubs, and which packages to install on them.
+
+#### Client initiated reporting
+
+A mode where you change the configuration so that the hub does not initiate connections to client hosts to fetch reports.
+Instead, the clients will establish a connection, and leave it open, until the hub is ready to use it to query for reporting data.
+Sometimes referred to as call collect.
+
+#### Configuration management database (CMDB)
 
-A Configuration Management Database.
 A term coined as part of the IT Infrastructure Library (ITIL) as an outgrowth of an inventory database.
 
-#### Commands
-#### Common control
+#### Code branch
+
+The development of software is a branching process.
+At certain times, the software code splits into different versions following different paths.
+Each path needs to be maintained separately for a while.
+This often happens when a release is made, because one wants to freeze the development of a public release (allowing nevertheless for some minor bugfixes), while continuing to add features to a branch leading to future versions.
+
 #### Components
 
 Standalone applications include `cf-agent`, `cf-promises`, `cf-runagent`, `cf-know`, `cf-report`, `cf-hub`
 
 Daemons include `cf-execd`, `cf-monitord`, and `cf-serverd`
 
+#### COSL license
+
+The Commercial Open Source License used for the CFEngine.
+
 #### Datatypes
 
 CFEngine's data types describe what a variable can contain.
 A variable can't be assigned a different type once it's been set.
-The commonly used data types are `string`, `slist`, `int`, `real`, and `data`.
+The commonly used data types are `string`, `slist` (string list), `int`, `real`, and `data`.
+
+#### Diff
+
+A `diff` is a report (originally that generated by the UNIX diff command) that details the differences between two files.
+The term is often used as slang meaning a file comparison.
 
-#### Directories
-#### Distribution
 #### Enterprise API
+
+The Enterprise API is a JSON HTTP REST API, allowing users to access CFEngine's functionality and reporting data programmatically.
+It can be used to generate reports, query data, create alerts, manage users, etc.
```

Comment:
Is this grammarly talking or you? :)

I am assuming grammarly says it's unclear because the previous sentence has 2 things the "it" could refer to (the API or CFEngine).

In this case I don't think it's an issue, and the suggested change hurts the reading flow.